pkg/auth/globalrole.go

```diff
@@ -65,9 +65,39 @@ func (g *GlobalRoleResolver) ClusterRulesFromRole(gr *v3.GlobalRole) ([]rbacv1.P
 		}
 		rules = append(rules, templateRules...)
 	}
+
 	return rules, nil
 }
 
+func (g *GlobalRoleResolver) FleetWorkspacePermissionsResourceRulesFromRole(gr *v3.GlobalRole) []rbacv1.PolicyRule {
```

Comment:
I don't fully understand what you mean here. These rules will be checked with the `FleetWorkspaceRulesResolver`, which means it'll check if the user have enough permission in all workspace except local (by looking at the `InheritedFleetWorkspacePermissions.ResourceRules` of the `GlobalRoles` he currently has)

Reply:
[Here](https://github.com/rancher/webhook/blob/4e47fcb77bd14e000b35fa486e9a1fd71f60994b/pkg/auth/globalrole.go#L51) is the relevant logic. Basically, even though RestrictedAdmin has no permissions in these fields, we need to give the appropriate permissions so that RA can give this role to others. But the RA doesn't have/doesn't use this field, so you need special logic here to handle that.

Reply:
done. I restricted the resources to just fleet resources, so RA cant grant core types in the local cluster. please have a look https://github.com/rancher/webhook/pull/348/commits/8525e3866f03bdb10c9d22556a98fa084678a7fe